addon/session-stores/cookie.js

```diff
@@ -159,33 +172,43 @@ export default BaseStore.extend({
   },
 
   _syncData() {
-    let data = this.restore();
-    if (!objectsAreEqual(data, this._lastData)) {
-      this._lastData = data;
-      this.trigger('sessionDataUpdated', data);
-    }
-    if (!Ember.testing) {
-      Ember.run.cancel(this._syncDataTimeout);
-      this._syncDataTimeout = Ember.run.later(this, this._syncData, 500);
-    }
+    return new Ember.RSVP.Promise((resolve) => {
+      this.restore().then((data) => {
+        if (!objectsAreEqual(data, this._lastData)) {
+          this._lastData = data;
+          this.trigger('sessionDataUpdated', data);
+        }
+        if (!Ember.testing) {
+          Ember.run.cancel(this._syncDataTimeout);
+          this._syncDataTimeout = Ember.run.later(this, this._syncData, 500);
+        }
+        resolve();
+      });
+    });
```

Comment:
Does this have to return a promise as well? It's actually synchronous and doesn't belong to the public API (which is only `persist`, `restore`, `clear`).